library/core/src/option.rs

```diff
@@ -1275,7 +1275,8 @@ impl<T, E> Option<Result<T, E>> {
     /// ```
     #[inline]
     #[stable(feature = "transpose_result", since = "1.33.0")]
-    pub fn transpose(self) -> Result<Option<T>, E> {
+    #[rustc_const_unstable(feature = "const_option", issue = "67441")]
```

Comment:
When this lands, we need to remember to add these to https://github.com/rust-lang/rust/issues/67441.

In fact it looks like `transpose` is already listed there, but `flatten` is missing.